code/modules/reagents/chem_splash.dm

```diff
@@ -4,75 +4,110 @@
 // Threatscale is a multiplier for the 'threat' of the grenade. If you're increasing the affected range drastically, you might want to improve this.
 // Extra heat affects the temperature of the mixture, and may cause it to react in different ways.
 
-
-/proc/chem_splash(turf/epicenter, affected_range = 3, list/datum/reagents/reactants = list(), extra_heat = 0, threatscale = 1, adminlog = 1)
+/**
+ * The basic chemical bomb proc.
+ * Combines a set of reagent holders into one holder and reacts it.
+ * If there are any reagents left over it spreads them across the surrounding environment.
+ *
+ * Arguments:
+ * - [epicenter][/turf]: The epicenter of the splash if some of the reagents aren't consumed.
+ * - [holder][/datum/reagents]: The holder to combine all of the reagents into. A temporary one is created if this is null.
+ * - [reactants][/list/datum/reagents]: The set of reagent holders to combine.
+ * - extra_heat: Some amount to heat the combined reagents by before reacting them.
+ * - threatscale: A multiplier for the reagent quantities involved.
+ * - adminlog: Whether to alert the admins that this has occured.
+ */
+/proc/chem_splash(turf/epicenter, datum/reagents/holder = null, affected_range = 3, list/datum/reagents/reactants = list(), extra_heat = 0, threatscale = 1, adminlog = 1)
 	if(!isturf(epicenter) || !reactants.len || threatscale <= 0)
 		return
-	var/has_reagents
-	var/total_reagents
-	for(var/datum/reagents/R in reactants)
-		if(R.total_volume)
-			has_reagents = 1
-			total_reagents += R.total_volume
+
+	var/has_reagents = FALSE
+	var/total_reagents = 0
+	for(var/datum/reagents/reactant in reactants)
+		if(reactant.total_volume)
+			has_reagents = TRUE
+			total_reagents += reactant.total_volume
 
 	if(!has_reagents)
-		return
+		return FALSE
 
-	var/datum/reagents/splash_holder = new/datum/reagents(total_reagents*threatscale)
-	splash_holder.my_atom = epicenter // For some reason this is setting my_atom to null, and causing runtime errors.
-	var/total_temp = 0
+	var/tmp_holder = null
+	if (isnull(holder))
+		tmp_holder = TRUE
+		holder = new /datum/reagents(total_reagents * threatscale)
+		holder.my_atom = epicenter
+	else
+		tmp_holder = FALSE
+		holder.maximum_volume += total_reagents * threatscale
```

Comment:
You might want to mention in the docstring this modification happens.

Reply:
It's undone at the end of the proc... but I see what you mean. Currently people might think the size of the original holder limits the size of the mix. Sure, I can do that.

Reply:
I've put a note about it in the docstring.